pkg/maps/ctmap/ctmap.go

```diff
@@ -618,9 +619,12 @@ func PurgeOrphanNATEntries(ctMapTCP, ctMapAny *Map) *NatGCStats {
 		} else if natKey.GetFlags()&tuple.TUPLE_F_OUT == tuple.TUPLE_F_OUT {
 			ingressCTKey := ingressCTKeyFromEgressNatKey(natKey)
 			egressCTKey := egressCTKeyFromEgressNatKey(natKey)
+			dsrCTKey := dsrCTKeyFromEgressNatKey(natKey)
 
-			if !ctEntryExist(ctMap, ingressCTKey) && !ctEntryExist(ctMap, egressCTKey) {
-				// No ingress and egress CT entries were found, delete the orphan egress NAT entry
+			if !ctEntryExist(ctMap, ingressCTKey) &&
```

Comment:
Q: once we get are fine to assume that DSR flows created by bpf_lxc no longer exist (probably in v1.15), should remove the `!ctEntryExist(ctMap, ingressCTKey)` condition? If yes, then maybe let's add a TODO code comment.

Reply:
Yep, let's add that as TODO step in the to-be-opened issue for deprecating the bpf_lxc-level DSR code :+1:

Reply:
-> https://github.com/cilium/cilium/issues/24207